test/emulator_backend.go

```diff
@@ -668,6 +687,14 @@ func (e *EmulatorBackend) Events(
 	)
 }
 
+// Moves the time of the Blockchain's clock, by the
+// given time delta, in the form of seconds.
+func (e *EmulatorBackend) MoveTime(timeDelta int64) {
+	e.clock.TimeDelta = timeDelta
```

Comment:
Instead of setting the value, this should probably add the `timeDelta` to the `e.clock.TimeDelta`?
e.g. Calling `blockchain.moveTime(by: 15)` twice should ideally move the time by 30s.

Reply:
Would be good to have a test for that as well 👍

Reply:
Great point @SupunS :100: , the scenario of accumulating deltas didn't cross my mind. I will certainly update it and add relevant test cases.

Reply:
One thing that just came to me, when tampering with time operations during testing, there should be also a way to undo such changes. If `blockchain.moveTime` is passed a negative `timeDelta`, then it should go back in time.

Reply:
Opened https://github.com/onflow/cadence/pull/2734 to address the moving back in time functionality. Also added relevant test cases in both repositories.

Reply:
https://github.com/onflow/cadence-tools/pull/196/commits/347e47521b88f5155915332c772d66dd6769107c time delta accumulation fix.